epitome/test/dataset_test.py

```diff
@@ -205,3 +215,40 @@ def test_reserve_validation_indices(self):
         joined_indices = np.concatenate([dataset.indices[Dataset.TRAIN_VALID], dataset.indices[Dataset.TRAIN]])
         joined_indices.sort()
         self.assertTrue(len(np.setdiff1d(joined_indices, old_indices)) == 0 and len(np.setdiff1d(old_indices, joined_indices)) == 0)
+
+    def test_get_data_dir(self):
+        # Test unspecified model to have default data dir path
+        assert self.dataset.data_dir == self.epitome_test_path
+
+        # Create new dataset with new undownloaded data path
+        epitome_test_data_dir = os.path.dirname(self.epitome_test_path)
+        assert EpitomeDataset.get_data_dir(data_dir=epitome_test_data_dir, assembly="hg38") == os.path.join(epitome_test_data_dir, "hg38")
+
+        # Should error because undownloaded data path doesn't have a specified assembly
+        default_data_dir = os.path.join(DEFAULT_EPITOME_DATA_PATH, "hg19")
+        assert EpitomeDataset.get_data_dir() == default_data_dir
+
+        # Should error because assembly isn't contained in S3 cluster
+        with pytest.raises(AssertionError):
+            EpitomeDataset.get_data_dir(assembly="fake_assembly")
+
+        # Should error because data_dir doesn't have required files & assembly isn't specified
+        data_dir = os.path.join(epitome_test_data_dir, "fake_dir")
+        with pytest.raises(AssertionError):
+            EpitomeDataset.get_data_dir(data_dir=data_dir)
+
+        # Pass because data_dir doesn't have required files & assembly isn't specified
+        data_dir = os.path.join(epitome_test_data_dir, "fake_dir")
+        assert EpitomeDataset.get_data_dir(data_dir=data_dir, assembly="test") == os.path.join(data_dir, "test")
+
+        # Still fails because data_dir isn't the absolute data path downloaded above.
+        data_dir = os.path.join(epitome_test_data_dir, "fake_dir")
+        with pytest.raises(AssertionError):
+            EpitomeDataset.get_data_dir(data_dir=data_dir)
+
+        # Pass because data_dir now has the required files
+        data_dir = os.path.join(os.path.join(epitome_test_data_dir, "fake_dir"), "test")
+        assert EpitomeDataset.get_data_dir(data_dir=data_dir) == data_dir
```

Comment:
can you test for the case if data_dir=None?

Reply:
what about the case where data_dir is set but assembly is not?

Reply:
Both cases are tested in lines 233 and 255